diff --git a/clang/lib/Lex/HeaderSearch.cpp b/clang/lib/Lex/HeaderSearch.cpp
--- a/clang/lib/Lex/HeaderSearch.cpp
+++ b/clang/lib/Lex/HeaderSearch.cpp
@@ -961,6 +961,9 @@
     HFI->Resolved = true;
     auto ExternalHFI = ExternalSource->GetHeaderFileInfo(FE);
 
+    if (FE->getUID() >= FileInfo.size())
+        FileInfo.resize(FE->getUID() + 1);
+
     HFI = &FileInfo[FE->getUID()];
     if (ExternalHFI.External)
       mergeHeaderFileInfo(*HFI, ExternalHFI);
@@ -993,6 +996,9 @@
       HFI->Resolved = true;
       auto ExternalHFI = ExternalSource->GetHeaderFileInfo(FE);
 
+      if (FE->getUID() >= FileInfo.size())
+          FileInfo.resize(FE->getUID() + 1);
+
       HFI = &FileInfo[FE->getUID()];
       if (ExternalHFI.External)
         mergeHeaderFileInfo(*HFI, ExternalHFI);
